system/src/chat/hooks.mjs

```diff
@@ -132,3 +132,146 @@ export default function onRenderChatMessage(app, html, data) {
+
+		return game.user.isGM
+			&& canvas.tokens?.controlled.length
+			&& (_chatMessageIsDamageCardSecondary(message));
+	};
+
+	options.push(
+		{
+			name: game.i18n.localize("SHADOWDARK.chat_card.context.apply_damage"),
+			icon: '<i class="fas fa-user-minus"></i>',
+			condition: canApplyDamage,
+			callback: li => applyChatCardDamage(li, 1),
+		},
+		{
+			name: game.i18n.localize("SHADOWDARK.chat_card.context.apply_healing"),
+			icon: '<i class="fas fa-user-plus"></i>',
+			condition: canApplyDamage,
+			callback: li => applyChatCardDamage(li, -1),
+		},
+		{
+			name: game.i18n.localize("SHADOWDARK.chat_card.context.apply_damage_secondary"),
+			icon: '<i class="fas fa-user-minus"></i>',
+			condition: canApplySecondaryDamage,
+			callback: li => applyChatCardDamageSecondary(li, 1),
+		},
+		{
+			name: game.i18n.localize("SHADOWDARK.chat_card.context.apply_healing_secondary"),
+			icon: '<i class="fas fa-user-plus"></i>',
+			condition: canApplySecondaryDamage,
+			callback: li => applyChatCardDamageSecondary(li, -1),
+		}
+	);
+	return options;
+}
+
+/**
+ * Apply rolled dice damage to the token or tokens which are currently controlled.
+ * The multipliers allows for damage to be scaled for healing, or other modifications
+ *
+ * @param {HTMLElement} li      The chat entry which contains the roll data
+ * @param {number} multiplier   A damage multiplier to apply to the rolled damage.
+ * @returns {Promise}
+ */
+function applyChatCardDamage(li, multiplier) {
+
+	const message = game.messages.get(li.data("messageId"));
+	let roll;
+
+	// There are two version of this check.
+	// Version #1, this is a single roll
+
+	if (_chatMessageIsBasicRoll(message)) {
+		roll = message.rolls[0];
+	}
+
+	if (_chatMessageIsDamageCard(message)) {
+		roll = message?.flags.rolls.primaryDamage.roll;
+	}
+
+	return Promise.all(canvas.tokens.controlled.map(t => {
+	  const a = t.actor;
+	  return a.applyDamage(roll.total, multiplier);
+	}));
+}
+
+/**
+ * Apply secondary rolled dice damage to the token or tokens which are currently controlled.
+ * The multipliers allows for damage to be scaled for healing, or other modifications.
+ * Specifically used for damage cards with two outputs, such at versatile.
+ *
+ * @param {HTMLElement} li      The chat entry which contains the roll data
+ * @param {number} multiplier   A damage multiplier to apply to the rolled damage.
+ * @returns {Promise}
+ */
+function applyChatCardDamageSecondary(li, multiplier) {
+
+	const message = game.messages.get(li.data("messageId"));
+
+	if (!_chatMessageIsDamageCardSecondary(message)) {
+		return;
+	}
+
+	let roll = message?.flags.rolls.secondaryDamage.roll;
+
+	return Promise.all(canvas.tokens.controlled.map(t => {
+	  const a = t.actor;
+	  return a.applyDamage(roll.total, multiplier);
+	}));
+}
+
+
+/**
+ * Identifies basic ChatMessage rolls like `/r d6`
+ * @param {ChatMessage} message
+ * @returns
+ */
+function _chatMessageIsBasicRoll (message) {
```

Comment:
Is your linting on and properly using the config? This is throwing a warning for me if I look at it in VSCodium at least.